src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +296,35 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() <= 0 || !Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            return;
+        }
+
+        e.setAmount(e.getAmount() * 2);
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity().getPlayer();
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause != DamageCause.VOID || !Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            return;
+        }
+
+        Location loc = e.getEntity().getPlayer().getBedSpawnLocation();
+        if (loc == null)
```

Comment:
Please add brackets here or preferabbly even a method `getSafeLocation()` with docs and annotations.

Reply:
Regarding this method, @TheBusyBiscuit, do I just make it inside of TalismanListener? Or somewhere in the api?

Reply:
I'd just put it in this file yeah.